lib/fastlane/plugin/wpmreleasetoolkit/actions/common/prototype_build_details_comment_action.rb

```diff
@@ -0,0 +1,259 @@
+                       elsif URI(url_or_emoji)
+                         url_or_emoji
+                       end
+        app_icon_url ? "<img alt='#{alt}' align='top' src='#{app_icon_url}' width='20px' />" : ''
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        'Generates a string providing all the details of a prototype build, nicely-formatted and ready to be used as a PR comment (e.g. via `comment_on_pr`).'
+      end
+
+      def self.details
+        <<~DESC
+          Generates a string providing all the details of a prototype build, nicely-formatted as HTML.
+          The returned string will typically be subsequently used by the `comment_on_pr` action to post that HTML as comment on a PR.
+
+          If you used the `appcenter_upload` lane (to upload the Prototype build to App Center) before calling this action, and pass
+          a value to the `app_center_org_name` parameter, then many of the parameters and metadata will be automatically extracted
+          from the `lane_context` provided by `appcenter_upload`, including:
+
+           - The `app_center_app_name`, `app_center_release_id` and installation URL to use for the QR code to point to that release in App Center
+           - The `app_icon`
+           - The app's Build Number / versionCode
+           - The app's Version / versionName
+           - The app's Bundle ID / Application ID
+           - A `footnote` mentioning the MC tool for Automatticians to add themselves to App Center
+
+          This means that if you are using App Center to distribute your Prototype Build, the only parameters you *have* to provide
+          to this action are `app_display_name` and `app_center_org_name`; plus, for `metadata` most of the interesting values will already be pre-filled.
+
+          Any of those implicit default values/metadata can of course be overridden by passing an explicit value to the appropriate parameter(s).
+        DESC
+      end
+
+      def self.available_options
+        app_center_auto = '(will be automatically extracted from `lane_context if you used `appcenter_upload` to distribute your Prototype build)'
+        [
+          FastlaneCore::ConfigItem.new(
+            key: :app_display_name,
+            env_name: 'FL_PROTOTYPE_BUILD_DETAILS_COMMENT_APP_DISPLAY_NAME',
+            description: 'The display name to use for the app in the comment message',
+            optional: false,
+            type: String
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :app_center_org_name,
+            env_name: 'APPCENTER_OWNER_NAME', # Intentionally the same as the one used by the `appcenter_upload` action
+            description: 'The name of the organization in App Center (if you used `appcenter_upload` to distribute your Prototype build)',
```

Comment:
Nope, in this case it was intentional not to use `{app_center_auto}` (aka "(will be automatically extracted from `lane_context if you used `appcenter_upload` to distribute your Prototype build)"), because that's not the case for this property.

In fact, the `app_center_org_name` is the only value that you have to provide explicitly if you want the action to use App Center.
 - Not only is this value not exposed in the `lane_context` filled by `appcenter_upload`, unlike all the other properties (the App Center API response includes the app name, but not the app org… who knows why), so this is NOT extracted from `lane_context` here
 - But also, I figured this was a great thing to have a way to tell the action to NOT assume that because `appcenter_upload` was used that means we want to use its `lane_context` implicitly (without any way of disabling that) for the comment generation.
    - For example, imagine a case where a fastlane lane (1) creates a beta build and uploads it App Center via `appcenter_upload` (2) and then creates a Prototype Build, which they do NOT upload to App Center but instead to S3. We don't want the `prototype_build_details_comment` lane to pick up the `lane_context` from the `appcenter_upload` call from (1) while we didn't use App Center for (2).
    - So in the end I decided to consider this fact of "`app_center_org_name` not being exposed in the `lane_context`" as a positive rather than a negative, as it allows you too use it as a signaling flag of it you want the action to consider you used App Center or not (instead not having a way to disable the `lane_context` inference) 🤷